src/k8s-engine/images/imageProcessor.ts

```diff
@@ -504,6 +504,69 @@ export abstract class ImageProcessor extends EventEmitter {
 
     console.log(`Installing kim: kim ${ args.join(' ') }`);
 
+    // Need two loops here. First loop is to verify that we're running a containerd node
+    // Then run the builder installer, and wait a bit for it to start up
+    while (true) {
+      try {
+        await childProcess.spawnFile(
+          resources.executable('kim'),
+          args,
+          {
+            stdio:       ['ignore', 'pipe', 'pipe'],
+            windowsHide: true,
+          });
+        break;
+      } catch (e) {
+        console.error(`Failed to restart the kim builder: ${ e.message }.`);
+        if (!e.stderr?.includes('Error: container runtime `docker` not supported')) {
+          console.error(`Attempt to run 'kim install builder' => error: '${ e.stderr }'`);
+          console.error('A reset might be necessary to support building images.');
+
+          return;
+        }
+      }
+
+      console.log('Waiting for the kubernetes backend to start using the newest nodes.');
+      if ((Date.now() - startTime) > maxWaitTime) {
+        console.log(`Waited more than ${ maxWaitTime / 1000 } secs; Probably a reset is needed to build images.`);
+        break;
+      }
+      await util.promisify(setTimeout)(waitTime);
+    }
+
+    // And now wait for the builder to be ready
+    while (true) {
+      if (await backend.isServiceReady('kube-image', 'builder')) {
+        console.log('Image-building support is now installed and running');
+        break;
+      }
+      if ((Date.now() - startTime) > maxWaitTime) {
+        console.log(`Waited more than ${ maxWaitTime / 1000 } secs; image-building might work fine later.`);
+        break;
+      }
+      await util.promisify(setTimeout)(waitTime);
+    }
+  }
+
+  /**
+   * Uninstall the kim backend builder, not needed for moby
+   * @param backend API to communicate with Kubernetes.
+   * @param address For the kim image processor, the end point address.
+   */
+  async uninstallKimBuilder(backend: K8s.KubernetesBackend, address?: string) {
+    if (!(await backend.isServiceReady('kube-image', 'builder', false))) {
+      console.log("Skipping kim uninstall: service isn't running");
```

Comment:
Not quite true — the service might be installed but unready.

Reply:
Obsolete code.